oop-sensor-data-analyzer/src/file/handling/JsonExporter.java

```diff
@@ -120,31 +124,36 @@ void writeSensorsWithHighestLatestReading(FileWriter fileWriter, List<AbstractSe
     }
 
     @Override
-    void writeLatestReading(FileWriter fileWriter, List<AbstractSensor> sensorList) throws IOException {
+    void writeLatestReading(FileWriter fileWriter, List<Sensor> sensorList) throws IOException {
         fileWriter.write("\"Sensors latest readings\": [\n");
         SensorType[] types = SensorType.values();
-        for (int i = 0; i < types.length; i++) {
-            SensorType type = types[i];
-            Map<SensorType, Reading> latestReadings = SensorAnalyzer.getLatestReadingsGroupedByType(sensorList, type);
+        boolean first = true;
 
-            fileWriter.write("    {\n");
-            fileWriter.write("    \"sensor type\" : " + "\"" + type + "\",\n");
-            fileWriter.write("       \"reading\" : {" + "\n");
 
-            //Map.Entry<> entry : xy.entrySet used for getting the Map 'Key' and the 'Value'!!
-            for (Map.Entry<SensorType, Reading> entry : latestReadings.entrySet()) {
-                SensorType key = entry.getKey();
-                Reading reading = entry.getValue();
+        Map<Sensor, Reading> latestReadings = SensorAnalyzer.getLatestReadingsGroupedByType(sensorList);
 
-                fileWriter.write("           \"latest\" : \"" + key + "\",\n");
-                fileWriter.write("           \"time\" : \"" + reading.getTimestamp().toString() + "\"\n");
-                fileWriter.write("        }\n");
-                fileWriter.write("  }");
-            }
-            if (i != types.length - 1) {
+        for (Map.Entry<Sensor, Reading> entry : latestReadings.entrySet()) {
+            Sensor key = entry.getKey();
+            Reading reading = entry.getValue();
+            String unit = getUnit(key.getSensorType());
+
+            if (!first) {
                 fileWriter.write(",\n");
+            } else {
+                first = false;
             }
+
+            fileWriter.write("  {\n");
+            fileWriter.write("    \"sensor type\" : " + "\"" + key.getSensorType() + "\",\n");
+            fileWriter.write("    \"sensor id\" : " + "\"" + key.getId() + "\",\n");
+            fileWriter.write("       \"reading\" : {" + "\n");
+            fileWriter.write("          \"value\" : " + reading.reading() + ",\n");
+            fileWriter.write("          \"unit\" : \"" + unit + "\",\n");
+            fileWriter.write("          \"time\" : \"" + reading.timestamp() + "\"\n");
+            fileWriter.write("        }\n");
+            fileWriter.write("  }");
```

Comment:
please note that whitespaces and enters are not necessarily the part of a valid json though (just letting you know)